src/core/qgsmaplayer.cpp

```diff
@@ -1166,7 +1167,7 @@ bool QgsMapLayer::importNamedStyle( QDomDocument &myDocument, QString &myErrorMe
   }
 #endif
 
-  return readSymbology( myRoot, myErrorMessage );
+  return readSymbology( myRoot, myErrorMessage, QgsPathResolver() ); // TODO: support relative paths?
```

Comment:
I guess this is more for just vector layers, e.g SVG symbols. I just focused on the TODO note and missed that this refers to the base class.